From 7bd8a45ced2e346a0092d052740a568b2c9f9974 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 29 Apr 2017 20:17:42 +0100
Subject: [PATCH] libdevcheck: Fix portability issues with GNU C

The libdevcheck component is explicitly dependent on GNU C extensions,
such as asprintf. Additionally with glibc, O_NOATIME is only exposed
when _GNU_SOURCE is defined.

As such, #define _GNU_SOURCE is the dependent components of libdevcheck
to ensure that this builds with compliance on GLibc using platforms.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 libdevcheck/copy.c              | 2 ++
 libdevcheck/posix_write_zeros.c | 2 ++
 libdevcheck/read_test.c         | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/libdevcheck/copy.c b/libdevcheck/copy.c
index c98e996..8e9f0cd 100644
--- a/libdevcheck/copy.c
+++ b/libdevcheck/copy.c
@@ -1,3 +1,5 @@
+#define _GNU_SOURCE
+
 #define _FILE_OFFSET_BITS 64
 #include <stdlib.h>
 #include <string.h>
diff --git a/libdevcheck/posix_write_zeros.c b/libdevcheck/posix_write_zeros.c
index 30e8c23..da16dab 100644
--- a/libdevcheck/posix_write_zeros.c
+++ b/libdevcheck/posix_write_zeros.c
@@ -1,3 +1,5 @@
+#define _GNU_SOURCE
+
 #define _FILE_OFFSET_BITS 64
 #include <stdio.h>
 #include <stdlib.h>
diff --git a/libdevcheck/read_test.c b/libdevcheck/read_test.c
index c42823e..63f1d2f 100644
--- a/libdevcheck/read_test.c
+++ b/libdevcheck/read_test.c
@@ -1,3 +1,5 @@
+#define _GNU_SOURCE
+
 #define _FILE_OFFSET_BITS 64
 #include <stdio.h>
 #include <stdlib.h>
-- 
2.12.2

